src/System.IO.Ports/tests/SerialPort/OpenDevices.cs

```diff
@@ -19,9 +19,12 @@ public void OpenDevices01()
             DosDevices dosDevices = new DosDevices();
             Regex comPortNameRegex = new Regex(@"com\d{1,3}", RegexOptions.IgnoreCase);
 
+            // This test enumerates all DosDevices and attempts to open them, expecting an exception.
+            // However, some non true Serial devices can successfully open!
+            // Added check in SerialPort.Open() to throw exception if not a valid PortName & Also added condition check for valid ports below
             foreach (KeyValuePair<string, string> keyValuePair in dosDevices)
             {
-                if (!string.IsNullOrEmpty(keyValuePair.Key) && !comPortNameRegex.IsMatch(keyValuePair.Key))
+                if (!string.IsNullOrEmpty(keyValuePair.Key) && !comPortNameRegex.IsMatch(keyValuePair.Key) && !keyValuePair.Key.Contains("86e0d1e0-8089-11d0-9ce4-08003e301f73"))
```

Comment:
this guid should be a constant somewhere instead of being inline everywhere (constant should have a comment where it's coming from)

Reply:
Thanks for input!
Yes agree!! But struggling trying to find a suitable location to place the Constant....
Any suggestions?

Reply:
anywhere, just make it internal. For tests you can define a duplicate constant